notebooks/extrap-with-metadata-aggregated.ipynb

```diff
@@ -0,0 +1,257 @@
+    "t_ens = th.Thicket.from_caliperreader(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Specifically, the metadata table for this set of profiles contains a `jobsize` column, which provides the amount of cores used for each profile."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "t_ens.metadata[\"jobsize\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. More Information on a Function\n",
+    "***\n",
+    "You can use the help() method within Python to see the information for a given object. You can do this by typing help(object). \n",
+    "This will allow you to see the arguments for the function, and what will be returned. An example is below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "help(Modeling)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Create Models\n",
+    "\n",
+    "First, we construct the `Modeling` object by passing all the relevant data to it. We provide `jobsize` as the `param_name` argument so the model will grab this column from the metadata table to use as our parameter. We also sub-select some metrics, since this dataset has a lot of metrics (otherwise the modeling will take a long time to do all metrics).\n",
+    "\n",
+    "Then, we call `produce_models` on that object (it's unecessary to provide an aggregation function since the data is already aggregated.).\n",
+    "\n",
+    "**NOTE:** For this example, you can view all the metric columns by adding a new cell and running: `t_ens.dataframe.columns`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mdl = Modeling(\n",
+    "    t_ens,\n",
+    "    \"jobsize\",\n",
+    "    chosen_metrics=[\n",
+    "        \"Avg time/rank\",\n",
+    "        \"Max time/rank\",\n",
+    "    ],\n",
+    ")\n",
+    "\n",
+    "mdl.produce_models()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Models Dataframe\n",
+    "\n",
+    "Model hypothesis functions are stored in the thicket's statsframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "t_ens.statsframe.dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Show the Models Dataframe with Embedded Plots\n",
+    "\n",
+    "(For every `node`, sub-selected `metric` combination)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "with pd.option_context('display.max_colwidth', 1):\n",
```

Comment:
use double quotes in code: `with pd.option_context("display.max_colwidth", 1):\n`